jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/actions/OpenLogStreamInEditor.kt

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.actions
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.ui.table.JBTable
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
+import software.aws.toolkits.resources.message
+
+class OpenLogStreamInEditor(
+    private val project: Project,
+    private val logGroup: String,
+    private val groupTable: JBTable
+) :
+    AnAction(message("cloudwatch.logs.open_in_editor"), null, AllIcons.Actions.Menu_open),
+    CoroutineScope by ApplicationThreadPoolScope("OpenLogStreamInEditor"),
+    DumbAware {
+    private val edt = getCoroutineUiContext(ModalityState.defaultModalityState())
+
+    override fun actionPerformed(e: AnActionEvent) {
+        launch {
+            actionPerformedSuspend(e)
+        }
+    }
+
+    private suspend fun actionPerformedSuspend(e: AnActionEvent) {
```

Comment:
Do we want to max file size this and offer to download to disk, that way large file editor still works?

Reply:
`factory.createFileFromText` has `noSizeLimit` set. I think we should add some kind of a limit but I'm not sure a good behavior we could do if we reach that limit. Saving it to a file could still have some kind of terrible mega log that wouldn't really work as a text file and I'm not sure what we should do in that situation